### PR TITLE
[config] Cleanup and SafetyRules prep

### PR DESCRIPTION
**Pull request by davidiw**
This PR will serve to major purposes:
- Eliminating cruft due to genesis being the source of truth (recent work) and cruft in general
- Moving appropriate data into a place that makes sense from a SafetyRules perspective

The specifics:
- Cleaning up dynamic-config-builder junk
- Moving consensus_keypair from consensus config to test config
- Eliminating consensus_peers from consensus_config
- Eliminating the need for network_peers for validator networks
- Generating a ValidatorSet during validator generation rather than network / consensus peer configs
- Deleting consensus peer config

Testing relied primarily on integration tests and leveraging the local dynamic run of docker images.

### CODE REVIEW

**Comment by davidiw** [comment]
@dmitri-perelman, thanks for the review!

Updated to include testing method. I fixed our local running of docker images.
The smoke tests are actually really good at testing these type of things as well.
I did try to use the cluster testing but hit some issues with the environment setup.

**Comment by aching** [comment]
Btw, looks like a nice cleanup, thanks!

File: config/src/config/consensus_config.rs [hunk 1]
```diff
@@ -1,69 +1,40 @@
...
         }
     }
 }
 
+impl ConsensusConfig {
+    pub fn set_data_dir(&mut self, data_dir: PathBuf) {
+        self.safety_rules.set_data_dir(data_dir);
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (davidiw) [reply config/src/config/consensus_config.rs]: That's true, I think eventually I'm just going to move safety rules to being top level.

**Pull request closed without merging**